flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/TaskExecutorResourceSpec.java

```diff
@@ -186,6 +192,21 @@ public MemorySize getJvmDirectMemorySize() {
 		return frameworkOffHeapMemorySize.add(taskOffHeapSize).add(shuffleMemSize);
 	}
 
+	public double getDefaultSlotFraction() {
+		return defaultSlotFraction;
+	}
+
+	public ResourceProfile getDefaultSlotResourceProfile() {
```

Comment:
I would not add one more concern of default profile to `TaskExecutorResourceSpec`.
`TaskExecutorResourceSpecUtils` could have a factory method for the default profile:
`ResourceProfile  generateDefaultSlotResourceProfile(config(, TaskExecutorResourceSpec))`